### PR TITLE
Adds support for hreflang, spelling fixes, performance updates

### LINKED ISSUE #152: Allow alternate links for hreflang / canonical
Hey, first of all thanks for your nice plugin!

I would need to tell the crawlers that a specific dynamic page is available in a different language:
https://developers.google.com/search/docs/advanced/crawling/localized-versions

```
<url xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    <loc>https://mypage.de</loc>
    <lastmod>2021-04-20T18:45:49+01:00</lastmod>
    <priority>1.0</priority>
    <xhtml:link xmlns:xhtml="http://www.w3.org/1999/xhtml" rel="alternate" hreflang="de" href="https://mypage.de/en" />
    <xhtml:link xmlns:xhtml="http://www.w3.org/1999/xhtml" rel="alternate" hreflang="es" href="https://mypage.de/es" />
  </url>
```

Is this possible with the current version? I have not found it in your `fields` example:

```
const fields = [
    {
      loc: 'https://example.com', // Absolute url
      lastmod: new Date().toISOString(),
      // changefreq
      // priority
    },
    {
      loc: 'https://example.com/dynamic-path-2', // Absolute url
      lastmod: new Date().toISOString(),
      // changefreq
      // priority
    },
  ]
```

### PR DESCRIPTION
**Pull request by EthanStandel**
This adds support for the hreflang tag (Fixes #152 ), so that your Sitemap can identify that it supports multiple languages to web-crawlers.

Documentation for that syntax is here:
https://developers.google.com/search/docs/advanced/crawling/localized-versions#sitemap

Other items
* The type `ISitemapField` was misspelled as `ISitemapFiled`.
* The constant rebuilding and the garbage collecting of strings when building XML can be very innefficient.  I converted this logic from `Array.reduce(...)` to `Array.map(...).join("")`

### CODE REVIEW

**Comment by iamvishnusankar** [comment]
@EthanStandel Thanks a lot for this PR! 🙏  

Funny that nobody noticed this spell error for this long 😄. Can you add the documentation regarding alternativeRefs in README.  Thanks!

**Comment by EthanStandel** [comment]
@iamvishnusankar Sure thing!  Always happy to help public projects 😄 

After going through the README a bit more in depth, I realized there there were some more necessary changes to allow alternateRefs for the global config.

There's now two new commits.  The first adds alternateRefs to the global config options and then sets the transformation function to just copy them and fully builds them out in `createUrlSet`.  The second commit documents these changes in the README as requested!

File: packages/next-sitemap/src/config/index.ts [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
 import {
   IConfig,
-  ISitemapFiled,
+  ISitemapField,
   IRuntimePaths,
   IExportMarker,
 } from '../interface'
 import { merge } from '@corex/deepmerge'
 import { loadFile } from '../file'
+import { absoluteUrl } from '../url'
```
Comment (iamvishnusankar) [inline packages/next-sitemap/src/config/index.ts:11]: @EthanStandel Linting errors

Comment (EthanStandel) [reply packages/next-sitemap/src/config/index.ts]: FIxed!

**Review by iamvishnusankar — APPROVED** [review]

**Comment by iamvishnusankar** [comment]
@EthanStandel Thanks a lot for this PR. Approved :)

**Pull request merged**